### PR TITLE
Refactor all examples using transforms

### PR DESCRIPTION
**Pull request by Ra1Nik**
Refactor all examples, using transforms implemented in https://github.com/lightly-ai/lightly/pull/1074.

### CODE REVIEW

File: examples/pytorch/mae.py [hunk 1]
```diff
@@ -79,11 +82,11 @@ def forward(self, images):
 pascal_voc = torchvision.datasets.VOCDetection(
     "datasets/pascal_voc", download=True, target_transform=lambda t: 0
 )
-dataset = LightlyDataset.from_torch_dataset(pascal_voc)
+dataset = LightlyDataset.from_torch_dataset(pascal_voc, transform=MAETransform())
 # or create a dataset from a folder containing images or videos:
 # dataset = LightlyDataset("path/to/folder")
 
-collate_fn = MAECollateFunction()
+collate_fn = MultiViewCollate()
```
Comment (philippmwirth) [inline examples/pytorch/mae.py:89]: We don't need the `MultiViewCollate` for `MAETransform` do we?

File: examples/pytorch/mae.py [hunk 2]
```diff
@@ -29,7 +30,7 @@ def __init__(self, vit):
             embed_input_dim=vit.hidden_dim,
             hidden_dim=decoder_dim,
             mlp_dim=decoder_dim * 4,
-            out_dim=vit.patch_size ** 2 * 3,
+            out_dim=vit.patch_size**2 * 3,
```
Comment (philippmwirth) [inline examples/pytorch/mae.py:33]: Why did you change formatting here?

File: examples/pytorch/mae.py [hunk 3]
```diff
@@ -101,7 +104,7 @@ def forward(self, images):
 for epoch in range(10):
     total_loss = 0
     for images, _, _ in dataloader:
-        images = images.to(device)
+        images = images[0].to(device)  # images is a list containing only one view
```
Comment (philippmwirth) [inline examples/pytorch/mae.py:107]: We should be able to drop this if we get rid of the `MultiViewCollate`

File: examples/pytorch/msn.py [hunk 4]
```diff
@@ -56,9 +58,9 @@ def forward_masked(self, images):
     mlp_dim=384 * 4,
 )
 model = MSN(vit)
-# # or use a torchvision ViT backbone:
+# # or use a torchvision ViT backbone:
```
Comment (philippmwirth) [inline examples/pytorch/msn.py:61]: Remove the double comment here.

File: examples/pytorch/msn.py [hunk 5]
```diff
@@ -56,9 +58,9 @@ def forward_masked(self, images):
...
 # vit = torchvision.models.vit_b_32(pretrained=False)
-# moel = MSN(vit)
+# moel = MSN(vit)
```
Comment (philippmwirth) [inline examples/pytorch/msn.py:63]: typo + double comment

File: examples/pytorch/msn.py [hunk 6]
```diff
@@ -21,7 +22,7 @@ def __init__(self, vit):
 
         self.mask_ratio = 0.15
         self.backbone = MAEBackbone.from_vit(vit)
-        self.projection_head = MSNProjectionHead()
+        self.projection_head = MSNProjectionHead(384)
```
Comment (philippmwirth) [inline examples/pytorch/msn.py:25]: Please use keyword arguments here.

File: examples/pytorch/nnclr.py [hunk 7]
```diff
@@ -41,11 +42,11 @@ def forward(self, x):
 memory_bank.to(device)
 
 cifar10 = torchvision.datasets.CIFAR10("datasets/cifar10", download=True)
-dataset = LightlyDataset.from_torch_dataset(cifar10)
+dataset = LightlyDataset.from_torch_dataset(cifar10, transform=SimCLRTransform())
```
Comment (philippmwirth) [inline examples/pytorch/nnclr.py:45]: You forgot the `input_size` here.

File: examples/pytorch_lightning/barlowtwins.py [hunk 8]
```diff
@@ -41,11 +42,13 @@ def configure_optimizers(self):
 model = BarlowTwins()
 
 cifar10 = torchvision.datasets.CIFAR10("datasets/cifar10", download=True)
-dataset = LightlyDataset.from_torch_dataset(cifar10)
+dataset = LightlyDataset.from_torch_dataset(
+    cifar10, transform=SimCLRTransform(input_size=32)
```
Comment (philippmwirth) [inline examples/pytorch_lightning/barlowtwins.py:46]: You're switching from `ImageCollateFunction` to `SimCLRTransform`. Is this intended?

File: examples/pytorch_lightning/msn.py [hunk 9]
```diff
@@ -90,11 +91,11 @@ def configure_optimizers(self):
 pascal_voc = torchvision.datasets.VOCDetection(
     "datasets/pascal_voc", download=True, target_transform=lambda t: 0
 )
-dataset = LightlyDataset.from_torch_dataset(pascal_voc)
+dataset = LightlyDataset.from_torch_dataset(pascal_voc, transform=MSNTransform())
```
Comment (philippmwirth) [inline examples/pytorch_lightning/msn.py:94]: In the pytorch example you set `MSNTransform(384)` but here you don't. What's the difference?

File: examples/pytorch_lightning/nnclr.py [hunk 10]
```diff
@@ -50,11 +51,11 @@ def configure_optimizers(self):
 model = NNCLR()
 
 cifar10 = torchvision.datasets.CIFAR10("datasets/cifar10", download=True)
-dataset = LightlyDataset.from_torch_dataset(cifar10)
+dataset = LightlyDataset.from_torch_dataset(cifar10, transform=SimCLRTransform())
```
Comment (philippmwirth) [inline examples/pytorch_lightning/nnclr.py:54]: Set the input size

File: examples/pytorch_lightning_distributed/msn.py [hunk 11]
```diff
@@ -92,11 +93,11 @@ def configure_optimizers(self):
 pascal_voc = torchvision.datasets.VOCDetection(
     "datasets/pascal_voc", download=True, target_transform=lambda t: 0
 )
-dataset = LightlyDataset.from_torch_dataset(pascal_voc)
+dataset = LightlyDataset.from_torch_dataset(pascal_voc, transform=MSNTransform())
```
Comment (philippmwirth) [inline examples/pytorch_lightning_distributed/msn.py:96]: Here, too: Do we have to set `MSNTransform(384)` or not?

File: examples/pytorch_lightning_distributed/nnclr.py [hunk 12]
```diff
@@ -50,11 +51,11 @@ def configure_optimizers(self):
 model = NNCLR()
 
 cifar10 = torchvision.datasets.CIFAR10("datasets/cifar10", download=True)
-dataset = LightlyDataset.from_torch_dataset(cifar10)
+dataset = LightlyDataset.from_torch_dataset(cifar10, transform=SimCLRTransform())
```
Comment (philippmwirth) [inline examples/pytorch_lightning_distributed/nnclr.py:54]: input size

File: lightly/data/dataset.py [hunk 13]
```diff
@@ -231,8 +232,11 @@ def from_torch_dataset(cls,
         )
 
         # populate it with the torch dataset
+        if transform is not None:
```
Comment (philippmwirth) [inline lightly/data/dataset.py:235]: Can you explain why this is necessary with a comment?

**Review by philippmwirth — CHANGES_REQUESTED** [review]
Thanks a lot for the changes!

File: examples/pytorch/mae.py (hunk 2 above)
Comment (Ra1Nik) [reply examples/pytorch/mae.py:33]: I simply run black

File: examples/pytorch_lightning/barlowtwins.py (hunk 8 above)
Comment (Ra1Nik) [reply examples/pytorch_lightning/barlowtwins.py:46]: Yes, as in the old collate `SimCLRTransform` was inheriting from `ImageCollateFunction` and just changing some parameters

File: examples/pytorch_lightning/msn.py (hunk 9 above)
Comment (Ra1Nik) [reply examples/pytorch_lightning/msn.py:94]: Here it was already set. The 384 wasn't in the transform but in the linear layer. It was forgotten and also the old implementation was not working.

File: examples/pytorch_lightning_distributed/msn.py (hunk 11 above)
Comment (Ra1Nik) [reply examples/pytorch_lightning_distributed/msn.py:96]: No, see above.

File: lightly/data/dataset.py [hunk 14]
```diff
@@ -234,6 +234,8 @@ def from_torch_dataset(cls,
         # populate it with the torch dataset
         if transform is not None:
             dataset.transform = transform
+            # If dataset is a VisionDataset dataset, we need to initialize the transform
```
Comment (philippmwirth) [inline lightly/data/dataset.py:237]: Let's change this to "If dataset is a VisionDataset, we need to initialize transforms, too."

File: examples/pytorch_lightning/msn.py (hunk 9 above)
Comment (philippmwirth) [reply examples/pytorch_lightning/msn.py:94]: Ah you're right.

File: examples/pytorch_lightning/barlowtwins.py (hunk 8 above)
Comment (philippmwirth) [reply examples/pytorch_lightning/barlowtwins.py:46]: Did you make sure the parameters are the same now as they were before or do we simply not care?

**Review by philippmwirth — COMMENTED** [review]
Some follow-up questions. Did you test this by running all the examples for 1 epoch?

**Comment by guarin** [comment]
@philippmwirth could you approve? I ran all the examples again (with PyTorch Lightning 2.0).
